### PR TITLE
feat: Add view_item_list event

### PR DESCRIPTION
**Pull request by ArthurTriis1**
Signed-off-by: Arthur Andrade <arthurfelandrade@gmail.com>

## What's the purpose of this pull request?
This PR add trigger for view_item_list analytics event, according to GA4.

## How it works? 
For this, when an collection page is visited the PR add in `window.dataLayer` one more event, with name and properties according to GA4.

## How to test it?
This PR also add cypress tests for this event, testing if the event is added to dataLayer and if the properties has been added too.

## References
[GA4](https://developers.google.com/analytics/devguides/collection/ga4/reference/events#view_item_list)

### CODE REVIEW

File: src/sdk/analytics/hooks/useIntersectionObserver.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import type { RefObject } from 'react'
+import { useEffect, useState } from 'react'
+
+interface Args extends IntersectionObserverInit {
+  freezeOnceVisible?: boolean
+}
+
+function useIntersectionObserver(
```
Comment (tlgimenes) [inline src/sdk/analytics/hooks/useIntersectionObserver.ts:8]: Why implementing this when we already use a lib for that?
https://github.com/vtex-sites/base.store/blob/6245be5b3eb7c63e3345e29960d9d0711d165ab1/src/sdk/search/PageBreak.tsx#L2

**Review by tlgimenes — CHANGES_REQUESTED** [review]

File: src/sdk/analytics/hooks/useIntersectionObserver.ts (hunk 1 above)
Comment (ArthurTriis1) [reply src/sdk/analytics/hooks/useIntersectionObserver.ts:8]: I didn't know this. I will use this lib.

File: src/sdk/analytics/hooks/useViewItemListEvent.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ProductSummary_ProductFragment[]
+  >([])
+
+  const [debouncedViewedProducts] = useDebounce(viewedProducts, DEBOUNCE_DELAY)
+
+  const {
+    currency: { code },
+  } = useSession()
+
+  const currency = code as CurrencyCode
```
Comment (heitorado) [inline src/sdk/analytics/hooks/useViewItemListEvent.ts:20]:
```suggestion
```

File: src/sdk/analytics/hooks/useViewItemListEvent.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  const event: ViewItemListEvent = useMemo(() => {
+    const products: Item[] =
+      debouncedViewedProducts.map((product) => {
+        return {
+          item_id: product.id,
+          item_name: product.name,
+          currency,
```
Comment (heitorado) [inline src/sdk/analytics/hooks/useViewItemListEvent.ts:28]:
```suggestion
          currency: code as CurrencyCode,
```

File: src/sdk/analytics/hooks/useViewItemListEvent.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }
+      }) ?? []
+
+    return {
+      type: 'view_item_list',
+      data: {
+        items: products,
+      },
+    }
+  }, [currency, debouncedViewedProducts])
```
Comment (heitorado) [inline src/sdk/analytics/hooks/useViewItemListEvent.ts:44]:
```suggestion
  }, [code, debouncedViewedProducts])
```

**Review by heitorado — COMMENTED** [review]
LGTM, just proposed some changes - passing forward the initial suggestion @icazevedo made on my last PR, because I saw the same code pattern on this one.

**Review by tlgimenes — APPROVED** [review]
LGTM 🎉

File: src/sdk/analytics/hooks/useViewItemListEvent.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  title: string
+}
+
+export const useViewItemListEvent = ({ products, title }: Props) => {
+  const {
+    currency: { code },
+  } = useSession()
+
+  const sendViewItemListEvent = useCallback(() => {
+    const productItens: Item[] = products.map((product) => {
```
Comment (icazevedo) [inline src/sdk/analytics/hooks/useViewItemListEvent.ts:17]:
```suggestion
    const productItems: Item[] = products.map((product) => {
```

File: src/sdk/analytics/hooks/useViewItemListEvent.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        price: product.offers.offers[0].price,
+      }
+    })
+
+    const event: ViewItemListEvent = {
+      type: 'view_item_list',
+      data: {
+        item_list_name: title,
+        item_list_id: title,
+        items: productItens,
```
Comment (icazevedo) [inline src/sdk/analytics/hooks/useViewItemListEvent.ts:36]:
```suggestion
        items: productItems,
```

**Review by icazevedo — APPROVED** [review]
Great! I just left a suggestion to fix a small typo on a variable's name.

**Pull request merged**